File: 01-node-ts/01-examples/02-post-with-params/src/api/controller.ts
import core from '../core/book';
import { Request, Response} from 'express';

let controller = {
    buyBook: (req: Request, res: Response) => {
        let bookId = req.body.bookId;
        let bookQuantity = req.body.bookQuantity;

        let purchaseInfo = core.buyBook(bookId, bookQuantity);

        res.setHeader('Content-Type', 'application/json');
        res.send(purchaseInfo);
    }
};

export default controller;

File: 04-clojure/01-examples/02-post-with-params/src/api/controller.clj
(ns api.controller
  (:require [core.books :as books]))

(defn buy-book
  [req]
  (let [id (get-in req [:body :bookId])
        quantity (get-in req [:body :bookQuantity])
        response (books/buy-book id quantity)]
    {:status 200 :body response}))
